File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        if(head==null){
            return null;
        }
        ListNode temp=head;
        ListNode prev=null;
        left=left-1;
        right=right-1;
        int count=0;
        while(temp!=null){
            if(count==left){
                break;
            }
            prev=temp;
            temp=temp.next;
            count=count+1;
        }
        ListNode left1=temp;
        while(temp!=null){
            if(count==right){
                break;
            }
            count=count+1;
            temp=temp.next;
        }
        ListNode right1=temp;
        ListNode next11=temp.next;
        ListNode curr=left1;
        ListNode prev1=next11;
        ListNode next=null;
        for(int i=left;i<=right;i++){
            next=curr.next;
            curr.next=prev1;
            prev1=curr;
            curr=next;
        }
        if(prev==null){
            head=right1;
        }else{
            prev.next=right1;

        }
        
        return head;
    }
}

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseBetween(struct ListNode* head, int left, int right) {
    if(left==right || head==NULL){
        return head;
    }
    int pos=1;
    struct ListNode* prev1=NULL;
    struct ListNode* temp=head;
    while(temp!=NULL && pos!=left){
        prev1=temp;
        temp=temp->next;
        pos++;
    }
    struct ListNode* p1=temp;
    struct ListNode* p2;
    struct ListNode* prev2=NULL;
    while(temp!=NULL && pos!=right){
        pos++;
        prev2=temp;
        temp=temp->next;
    }
    p2=temp;
    struct ListNode* prev=p2->next;
    
    for(int i=left;i<=right && p1!=NULL ;i++){
        struct ListNode* v=p1->next;
       
        p1->next=prev;
         prev=p1;
         p1=v;

    }
    if (prev1 != NULL)
        prev1->next = prev;
    else
        head = prev; 


     return head;
    
    

    
}